### PR TITLE
feat(STADTPULS-368): New Sensor Modal

### PR DESCRIPTION
**Pull request by vogelino**
This PR adds a Sensor modal component for the addition, edition, and deletion of a sensor.

### CODE REVIEW

File: src/components/EditAddSensorModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+        </fieldset>
+        <Controller
+          name='description'
+          control={control}
+          defaultValue={defaultValues?.description}
+          render={({ field }) => (
+            <FormTextarea
+              {...field}
+              label='Beschreibung'
+              placeholder='Beschreibe kurz der Sensor'
```
Comment (dnsos) [inline src/components/EditAddSensorModal/index.tsx:188]: Typo: der -> den

File: src/components/EditAddSensorModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+              defaultValue={defaultValues?.integration || integration}
+              render={({ field }) => (
+                <FormListBox
+                  {...field}
+                  onChange={newValue => {
+                    field.onChange(newValue);
+                    setIntegration(newValue as "http" | "ttn");
+                  }}
+                  label='Integration'
+                  placeholder='Wie möchtest du dein Sensor integrieren?'
```
Comment (dnsos) [inline src/components/EditAddSensorModal/index.tsx:235]: Typo: dein -> deinen

File: src/components/EditAddSensorModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+            <Controller
+              name='integration'
+              control={control}
+              defaultValue={defaultValues?.integration || integration}
+              render={({ field }) => (
+                <FormListBox
+                  {...field}
+                  onChange={newValue => {
+                    field.onChange(newValue);
+                    setIntegration(newValue as "http" | "ttn");
```
Comment (dnsos) [inline src/components/EditAddSensorModal/index.tsx:232]: Isn't it possible to set this to `as definitions["sensors"]["connection_type"]`? This way we don't have to manually update this once the `connection_type`s change

File: src/components/FormListBox/FormSelect.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,188 @@
+import { action } from "@storybook/addon-actions";
+import { Story, Meta } from "@storybook/react";
+import { useForm } from "react-hook-form";
+import { FormListBox, SelectOptionType } from ".";
+import ArrowBidirectionalVertical from "../../../public/images/icons/16px/arrowBidirectionalVertical.svg";
+import ArrowOutOfDoor from "../../../public/images/icons/16px/arrowOutOfDoor.svg";
+import Funnel from "../../../public/images/icons/16px/funnel.svg";
+
+export default {
+  title: "Forms/FormListBox",
```
Comment (dnsos) [inline src/components/FormListBox/FormSelect.stories.tsx:10]: Great that this one is so flexible, to handle icons, default selections etc. 👍

File: src/components/FormListBox/index.tsx [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+      <div className={`${className} mb-2 relative`}>
+        <Listbox
+          value={options.find(({ value }) => value === selectValue)}
+          onChange={newVal => {
+            if (typeof onChange === "function") onChange(newVal?.value);
+            setSelectValue(newVal?.value);
+          }}
+        >
+          {label && (
+            <Listbox.Label className='block mb-2 cursor-pointer transition hover:opacity-60'>
```
Comment (dnsos) [inline src/components/FormListBox/index.tsx:63]: So this is how we use `headless-ui`? It gives us a completely unstyled element and we use Tailwind to style it?

File: src/components/SensorPageHeader/withData.tsx [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+            "left-1/2 transform -translate-x-1/2 backdrop-filter backdrop-blur-md",
+          ].join(" ")}
+        >
+          {error && (
+            <Alert
+              type='error'
+              title='Fehler!'
+              message={
+                <>
+                  Es ist beim editieren ein Fehler aufgetreten:{" "}
```
Comment (dnsos) [inline src/components/SensorPageHeader/withData.tsx:89]: Maybe better: "Beim Editieren ist ein Fehler aufgetreten"

File: src/components/SensorPageHeader/withData.tsx [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+                variant='secondary'
+                onClick={() => setDeletionConfirmationIsOpened(false)}
+              >
+                Abbrechen
+              </Button>
+            </>
+          }
+        >
+          <div className='prose max-w-none'>
+            <p>Du bist dabeit der Sensor „{mergedSensor.name}“ zu löschen.</p>
```
Comment (dnsos) [inline src/components/SensorPageHeader/withData.tsx:194]: Typos, should be: Du bist dabei, den Sensor [...] zu löschen.

**Review by dnsos — CHANGES_REQUESTED** [review]
Great work @vogelino!  This really seemed like a complex one. I tested the interaction and it works very nicely.

I have a few comments and questions and one display problem:

Screenshot 2021-09-23 at 11 15 10

The tooltip turns slightly transparent and is outside of the viewport. The latter is of course a problem with the tooltip itself, not this PR.

File: src/components/EditAddSensorModal/index.tsx (hunk 3 above)
Comment (vogelino) [reply src/components/EditAddSensorModal/index.tsx:232]: The problem is that the backend currently also accepts "other" but the frontend doesn't. I will omit "other".

Comment (vogelino) [reply src/components/EditAddSensorModal/index.tsx:232]: I tried omit but it isn't possible on enums... :(

File: src/components/FormListBox/index.tsx (hunk 5 above)
Comment (vogelino) [reply src/components/FormListBox/index.tsx:63]: Exactly, and it takes care of all the aria labels, keyboard navigation, etc. Very cool!

File: src/components/SensorPageHeader/withData.tsx (hunk 7 above)
Comment (vogelino) [reply src/components/SensorPageHeader/withData.tsx:194]: OMG all the typo... Sorry. Danke :D

**Comment by vogelino** [comment]
@dnsos I fixed the opacity of the tooltip (not the position, I would do that in another branch) and added the focus on the name input on mount. That should do the trick. Let me know what you think?

**Review by dnsos — APPROVED** [review]
Good changes, thank you! 👍 

(I also think the tooltip positioning should be fixed in another task)

**Pull request merged**